test/unit/json_schema/resolve_path_test.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:json_schema/json_schema.dart';
+import 'package:test/test.dart';
+
+main() {
+  JsonSchema fooSchema;
+  setUp(() async {
+    fooSchema = await JsonSchema.createAsync({
+      '\$defs': {
+        'a': {
+          'const': 'found in ref',
+          'deeper': {'const': 'deeper in the schema'},
+          '\$ref': '#/\$defs/b'
+        },
+        'b': {'const': 'b in not resolved'}
+      },
+      'properties': {
+        'foo': {'\$ref': '#/\$defs/a'},
+        'baz': {
+          '\$ref': '#/\$defs/a',
+          'findMe': {'const': 'is found'}
```

Comment:
did you mean to nest this under `properties`?

Reply:
I did not mean to nest this under `properties`.